File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, char* argv[])
{

    if (argc == 2)
    {
        int inputKey = atoi(argv[1]);
        if(inputKey!=0)
        {
            string plaintext = get_string("plaintext: ");

            printf("ciphertext: ");
            for(int i=0;i<strlen(plaintext);i++)
            {
                if((plaintext[i] >= 'a' && plaintext[i] <= 'z') || (plaintext[i] >= 'A' && plaintext[i] <= 'Z') )
                {
                    if(isupper(plaintext[i]))
                    {
                        int c = (plaintext[i] - 65 + inputKey) % 26;
                        printf("%c", (c+65));
                    }else if(islower(plaintext[i]))
                        {
                        int c = (plaintext[i] - 97 + inputKey) % 26;
                        printf("%c", (c+97));
                        }
                }else
                    return 1;
            }
            printf("\n");
        }else
            printf("Usage: ./caesar key\n");
            return 2;
    }else
        printf("Usage: ./caesar key\n");
        return 3;
}

File: pset6/python/caesar/caesar.py
import sys
from cs50 import get_string

def main():

    if len(sys.argv) is not 2:
        print("Usage: python caesar.py key")
        return 1

    if sys.argv[1] is 0:
        print("Usage: python caesar.py key")
        return 2

    print("This is the name of the script: ", sys.argv[0])
    print("This is the keysize: ", sys.argv[1])
    print("Number of arguments: ", len(sys.argv))
    print("The arguments are: " , str(sys.argv))
    plaintext = get_string("plaintext: ")
    inputKey = sys.argv[1];

    for i in range(len(plaintext)):
        if (plaintext[i] >= 'a' and plaintext[i] <= 'z') or (plaintext[i] >= 'A' and plaintext[i] <= 'Z'):
            if str.isupper(plaintext[i]):
                c = (ord(plaintext[i]) - 65 + int(inputKey)) % 26
                print(chr(c+65), end="")
            elif str.islower(plaintext[i]):
                c = (ord(plaintext[i]) - 97 + int(inputKey)) % 26
                print(chr(c+97), end="")
        else:
            return 3

    print()



if __name__ == "__main__":
    main()